src/backend/distributed/utils/ruleutils_10.c

```diff
@@ -7134,6 +7134,12 @@ get_from_clause_item(Node *jtnode, Query *query, deparse_context *context)
 			/* Else print column aliases as needed */
 			get_column_alias_list(colinfo, context);
 		}
+		/* check if column's are given aliases in distributed tables */
+		else if (colinfo->parentUsing != NIL)
+		{
+			Assert(colinfo->printaliases);
+			get_column_alias_list(colinfo, context);
```

Comment:
I was wondering why we need the `if (GetRangeTblKind(rte) != CITUS_RTE_SHARD)` condition at all, but that seems to be for merge tables during re-partition joins. The deparsing logic would add more column aliases than there are columns in the merge table. It seems like there is a bug in the physical planner that we're trying to work around here, but I couldn't find a quick fix.

Making an exception for USING seems acceptable, since re-partition joins never have USING.